### DISCUSSION TITLE
Best practice to develop Web-UI

### DISCUSSION THREAD

**Discussion by djs0109** [micropython/micropython#9149] [General]
Hello everyone,

I am trying to develop a simple Web-UI for ESP32. I want the users to be able to configure the WLAN connection and do some simple interactions with the device. I have seen some examples [here](https://randomnerdtutorials.com/esp32-esp8266-micropython-web-server/) and [here](https://github.com/tayfunulu/WiFiManager/blob/master/wifimgr.py). These examples both use plain HTML and the request handling is based on string processing. Therefore, I would like to know, if there is a more recommendable way to develop Web-UI with Micropython.

Thanks!

**Comment by ThinkTransit** [comment]
Have a look at [microdot](https://github.com/miguelgrinberg/microdot) this will provide a more robust way of serving pages and managing post data much easier than string parsing.

**Reply by djs0109** [reply]
That looks really helpful!
